rr_ml/nodes/semantic_segmentation/run.py

```diff
@@ -0,0 +1,91 @@
+import time
+
+import keras
+import numpy as np
+import cv2
+import tensorflow as tf
+
+import rospy
+from cv_bridge import CvBridge, CvBridgeError
+from sensor_msgs.msg import Image
+
+from . import model_utils
+
+
+last_msg = None
+disable_gpu = True
+
+
+def image_msg_callback(msg):
+    global last_msg
+    last_msg = msg
+
+
+def main():
+    global last_msg
+
+    config = tf.ConfigProto()
+    if disable_gpu:
+        config.device_count['GPU'] = 0  # disables GPU if testing system has one
+    keras.backend.set_session(tf.Session(config=config))
+
+    rospy.init_node('segnet_labeler')
+
+    path_to_model = rospy.get_param('~model_path')
+    image_topic = rospy.get_param('~image_topic')
+    detection_topic = rospy.get_param('~detection_topic')
+
+    detect_pub = rospy.Publisher(detection_topic, Image, queue_size=1)
+    visualizer_pub = rospy.Publisher("~detect_vis", Image, queue_size=1)
+
+    rospy.Subscriber(image_topic, Image, image_msg_callback, queue_size=1, buff_size=10**8)
+
+    unet_helper = model_utils.UNetModelUtils()
+    cv_bridge = CvBridge()
+
+    model = keras.models.load_model(path_to_model)
+
+    while not rospy.is_shutdown():
+        if last_msg is None:
+            time.sleep(0.01)
+            continue
+
+        try:
+            # CvBridge makes img read-only without copy
+            cam_img = cv_bridge.imgmsg_to_cv2(last_msg, desired_encoding="bgr8").copy()
+        except CvBridgeError as e:
+            print e
+            continue
+        else:
+            last_msg = None
+
+        img = unet_helper.crop(cam_img)
+        cropped_height, cam_width = img.shape[:2]
+
+        X = unet_helper.prepare_input(img)
+
+        t0 = time.time()
+        y = model.predict_on_batch(X[np.newaxis])[0]
+        t1 = time.time()
+        print "neural net latency:", (t1 - t0)
```

Comment:
maybe consider doing this as a moving average if you're going to consider using latency as a metric to measure. If you no longer need/use latency information, maybe print it out with ros_debug so you're not getting too much output.